### PR TITLE
Refactor buildingNameToIndex

### PR DESCRIPTION
**Pull request by theis999**
Renamed (building)Index to type
Renamed (building)OrientationIndex to orientation
Renamed BuildingNameToIndex to BuildingNameToType

Changed BuildingNameToType to not be const, allowing the syntax BuildingNameToType[name]. Which i find looks nicer.

Added vector BuildingNames,
Changed FindBuildingName to use BuildingNames for instant lookup

### CODE REVIEW

File: src/Building.h [hunk 1]
```diff
@@ -6,14 +6,14 @@ struct Building
 {
 	double X;
 	double Y;
-	int Index;
-	Orientation OrientationIndex;
+	int type;
```
Comment (MortenTobiasNielsen) [inline src/Building.h:9]: This is a little confusing. This is not a type it is an integer.

File: src/BuildingNameToIndex.cpp [hunk 2]
```diff
@@ -2,13 +2,6 @@
...
-	for (auto it = BuildingNameToIndex.begin(); it != BuildingNameToIndex.end(); ++it)
-	{
-		if (it->second == index)
-		{
-			return it->first;
-		}
-	}
-
-	return "N/A";
+	if (index < 1 || index > BuildingNames.size()) return "N/A";
```
Comment (MortenTobiasNielsen) [inline src/BuildingNameToIndex.cpp:5]: Please use the same code style as the rest of the code base. :)

File: src/BuildingNameToIndex.h [hunk 3]
```diff
@@ -21,7 +29,108 @@ enum Assemblers
 	Centrifuge,
 };
 
-static const std::map<string, int> BuildingNameToIndex =
+static inline vector<string> BuildingNames = {
+	"NONE", // for alignment
```
Comment (MortenTobiasNielsen) [inline src/BuildingNameToIndex.h:33]: How is this supposed to be used?

**Review by MortenTobiasNielsen — COMMENTED** [review]
Just have a few comments. :)

File: src/Building.h (hunk 1 above)
Comment (theis999) [reply src/Building.h:9]: Index was an integer that referenced which type of building it was, not which number of building it was. 
Effectively it is an enumeration, so naming the variable type is appropriate.

File: src/BuildingNameToIndex.h (hunk 3 above)
Comment (theis999) [reply src/BuildingNameToIndex.h:33]: We need this to be true: `BuildingNames[index] = name -> BuildingNameToType[name] = index`
But the first element in BuildingNameToType has the value index=1, so the array BuildingNames needed to be padded 1 in the start.

**Pull request merged**